Okteto 1.14 release notes (#515)
okteto/docs: src/content/self-hosted/install/releases.mdx

id: releases
---

> ## 1.14.0

Reviewer: Can we follow the same sections' ordder than other releases? I think it is:
- Features
- Improvements
- Bugfixes

[DATE]

This version is compatible with Kubernetes versions 1.24 to 1.27.

### Deprecatation Notice

> - As of this versions deploying helm charts through the Okteto UI has been deprecated.

Reviewer 2: We are nor deprecating it, we are removing it (we no longer support the `application` key). We should probably document it as Breaking change as any potential instance using the key will no longer be able to deploy charts from the helm catalog.

@rlamana @viddo What do you think is the best way of communicating this?

Reviewer 3: @bigash42 FYI in case we need to prepare any additional comms

Reviewer 4: +1, I'd suggest:
```suggestion
- Helm charts catalog is no longer available in the Okteto UI. For alternative approach to deploy Helm charts see https://www.okteto.com/blog/making-your-helm-packaged-applications-ready-for-cloud-native-development-with-okteto/
```

Reviewer 2: Just a note. This wasn't available by default in SH since some time ago. What we completely removed is the possibility of enabling it (https://www.okteto.com/docs/self-hosted/administration/configuration/#applications).

Mentioning it because referring only to Okteto UI might be misleading. Also, we have to be very clear this is nothing to do with the Catalog feature.

> - Private Registry credentials defined in helm has been deprecated. Private Registry Credentials are now supported natively by Okteto and credentials defined in helm will be automatically migrated. More info [here](ADD_LINK)

Reviewer: @maroshii @ifbyol my understanding is that there is a migration guide in this case. Something like:
- upgrade
- remove `privateRegistry`
- upgrade again
And we could link to the CRDs doc too

Reviewer 2: Yes, but the PR is not merged yet: https://github.com/okteto/docs/pull/509. That's why there is a placeholder for the link. As soon as we merge the other, we will update this with the link to that guide

Reviewer 2: Link for the migration guide: https://www.okteto.com/docs/1.14/self-hosted/install/upgrade/#upgrading-to-okteto-114x.

We have to remove the `1.14`. from the docs so we always way to latest version

> ### Bug Fixes

Reviewer: We are using **Bugfixes** in other versions

Reviewer 2: Yes, and we have to change the one for the `1.13` release to match also with `Bugfixes`

- Reduce file permission for okteto-ca-pemstore certificate file in jobs  <!-- 7303 -->

> - Reduce expiration threshold for trial licenses from 60 to 7 days <!-- 7314 -->

Reviewer 2: This might be misleading. The first time I read it I understood that the trials duration is now 7 days.

What about something like: `Warning message for licenses close to expire is displayed now 7 days before expiration instead of 60 for trial licenses`?

cc @omnomagonz

Author: If we are gonna have the scope of the change be just about the warning I would just skip the entry completely.

### Improvements

- Upgrade ingress-nginx to 1.8.2 (Helm chart version 4.7.2)  <!-- 7295 -->

> - The Okteto CLI used for Pipeline installation has been upgraded to version [2.21.0](https://github.com/okteto/okteto/releases/tag/2.21.0) <!-- 7310 -->

Reviewer: I think we were documenting Okteto CLI upgrades as **Features**

> ### New Features

Reviewer: We are using **Features**. in other versions

> ### New Features

Reviewer 2: We should include the support for `1.27` of Kubernetes. Even we mention the supported Kubernetes version in the first part of the release notes, I think we should include it in features when a new version is started to be supported

id: releases
---

## 1.14.0

> [DATE]

Reviewer 2: Remember to change this (added just to avoid to forget about it)

> [DATE]

Author: ```suggestion
25 October 2023
```

This version is compatible with Kubernetes versions 1.24 to 1.27.

### Deprecation Notice

- Helm charts catalog is no longer available in the Okteto UI. For alternative approach to deploy Helm charts see https://www.okteto.com/blog/making-your-helm-packaged-applications-ready-for-cloud-native-development-with-okteto/

> - Private Registry credentials defined in helm has been deprecated. Private Registry Credentials are now supported natively by Okteto and credentials defined in helm will be automatically migrated. More info [here](ADD_LINK)

Author: ```suggestion
- Private Registry credentials defined in helm has been deprecated. Private Registry Credentials are now supported natively by Okteto and credentials defined in helm will be automatically migrated. More info [here](https://www.okteto.com/docs/self-hosted/install/upgrade/#upgrading-to-okteto-114x)
```

> - Private Registry credentials defined in helm has been deprecated. Private Registry Credentials are now supported natively by Okteto and credentials defined in helm will be automatically migrated. More info [here](https://www.okteto.com/docs/self-hosted/install/upgrade/#upgrading-to-okteto-114x)

Reviewer 3: ```suggestion
- Defining Private Registry credentials in Helm has been deprecated. Private Registry Credentials are now natively supported by Okteto and configurable in the Okteto UI. Credentials defined in Helm will be automatically migrated. More info [here](https://www.okteto.com/docs/self-hosted/install/upgrade/#upgrading-to-okteto-114x)
```

- Support for Kubernetes [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md) has been deprecated and will be removed in the next release.

### Features

> - Expose prometheus node metrics data in `/metrics`. More info [here](ADD_LINK).

Author: https://www.okteto.com/docs/administration/okteto-insights/

> - Expose prometheus node metrics data in `/metrics`. More info [here](ADD_LINK).

Author: ```suggestion
- Expose prometheus node metrics data in `/metrics`. More info [here](https://www.okteto.com/docs/administration/okteto-insights).
```

> - Native support for Private Registry Credentials. More info [here](ADD_LINK).

Reviewer 2: This could be the link for this: https://www.okteto.com/docs/1.14/administration/dashboard/#registry-credentials

(We would have to remove `1.14` from the path so we always way to current version)

> - Native support for Private Registry Credentials. More info [here](ADD_LINK).

Author: ```suggestion
- Native support for Private Registry Credentials. More info [here](https://www.okteto.com/docs/administration/dashboard/#registry-credentials).
```

> - Expose prometheus node metrics data in `/metrics`. More info [here](https://www.okteto.com/docs/administration/okteto-insights).

Reviewer 3: ```suggestion
- Added a new `/metrics` endpoint to expose prometheus node metrics data. More info [here](https://www.okteto.com/docs/administration/okteto-insights).
```

> - Native support for Private Registry Credentials. More info [here](https://www.okteto.com/docs/administration/dashboard/#registry-credentials).

Reviewer 3: ```suggestion
- Added native support for Private Registry Credentials, configurable directly in the Okteto UI. More info [here](https://www.okteto.com/docs/administration/dashboard/#registry-credentials).
```
